terraform/modules/hetzner-kubernetes/machines.resources.tf

```diff
@@ -0,0 +1,54 @@
+resource "hcloud_server" "machines" {
+  for_each = { for m in var.machines: "${m.group_name}-${m.machine_id}" => m }
+
+  name        = "${var.cluster_name}-${each.key}"
+  location    = var.default_location
+  image       = var.default_image
+  server_type = lookup(each.value, "machine_type", var.default_server_type)
+  ssh_keys    = var.ssh_keys
+
+  # NOTE: string is the only accepted type
+  # DOCS: for possible characters, see https://docs.hetzner.cloud/#labels
+  labels = merge(
+    {
+      cluster = var.cluster_name
+      group_name = each.value.group_name
+      machine_id = each.value.machine_id
+    },
+    contains( each.value.component_classes, "controlplane" ) ? { etcd_member_name = "etcd-${ each.value.machine_id }" } : {},
+    { for class in each.value.component_classes : "component-class.${local.LABEL_PREFIX}/${class}" => true }
+  )
+}
+
+
+resource "hcloud_server_network" "snw" {
+  for_each = toset([ for m in var.machines: "${m.group_name}-${m.machine_id}" ])
+
+  server_id = hcloud_server.machines[each.key].id
+  subnet_id = hcloud_network_subnet.sn.id
+}
+
+
+resource "hcloud_volume" "volumes" {
+  for_each = {
```

Comment:
What do we need statically defined volumes for? If we need storage in kubernetes, we should do that through `PersistentVolume` abstraction and the [csi-driver](https://github.com/hetznercloud/csi-driver).

Reply:
That was a [FUTUREWORK](https://github.com/wireapp/wire-server-deploy/blob/develop/terraform/modules/hetzner-kubernetes/main.tf#L1) I just tried to address.

Reply:
As we're already experiencing disk latency issues on the local disks on hetzner every once in a while (or at least that's the running theory) I'm not sure it's a good idea to run the etcd data directory on a network disk .

Usually replacing an etcd node with an empty node will just work; as it will backfill from the other server nodes.   (When the data is large; it can help copying over data from the old node first before killing it; and then copying it to the new node; but we can do that manually)